arch/x86/Kconfig

```diff
@@ -192,12 +192,38 @@ config X86_MMU
 
 config X86_MMU_PAGE_POOL_PAGES
 	int "Number of pages to reserve for building page tables"
-	default 17
+	default 0
 	depends on X86_MMU
 	help
-	  Building page tables at boot requires a pool of free memory pages
-	  to construct it. This can't be derived at build time, tune this
-	  to your SoC's specific memory map.
+	  Define the number of pages in the pool used to allocate page table
+	  data structures at runtime.
+
+	  Pages might need to be drawn from the pool during memory mapping
+	  operations, unless the address space has been completely pre-allocated.
+
+	  Pages will need to drawn from the pool to initialize memory domains.
+	  This does not include the default memory domain if KPTI=n.
+
+	  The specific value used here depends on the size of physical RAM,
```

Comment:
Tuning this is not intuitive. We can do better. I opened https://github.com/zephyrproject-rtos/zephyr/issues/29526